cmd/protoc-gen-go-grpc/grpc.go

```diff
@@ -493,6 +496,21 @@ func genServerMethod(gen *protogen.Plugin, file *protogen.File, g *protogen.Gene
 	return hname
 }
 
+// genStandaloneComments prints all leading comments for a FileDescriptorProto
+// location identified by the field number n.
+// Grabbed from: https://github.com/protocolbuffers/protobuf-go/blob/master/cmd/protoc-gen-go/internal_gengo/main.go#L110
```

Comment:
This will require the appropriate license requirements to be satisfied.  It would be better if you could reimplement this functionality yourself so it's not necessary.

Reply:
I made the PR hoping you all could suggest a better way, based on your knowledge of the code. 

I don't care if my changes make it in, I just want the generated files to include the header comments from the .proto file.

Reply:
If your changes don't make it in, the generated files will not include the header comments. 😄  Our team doesn't have the resources / priority to take on this particular task right now, but we can do code reviews.  And we cannot accept code used under other licenses in our repo at this time.

Reply:
I am very much not a copyright lawyer, but I can state with confidence that this function was written by at-the-time Googler Joe Tsai in https://go.dev/cl/189238 (with some modifications in further CLs, also written by Joe).

Reply:
> I am very much not a copyright lawyer, but I can state with confidence that this function was written by at-the-time Googler Joe Tsai in go.dev/cl/189238 (with some modifications in further CLs, also written by Joe).

My concern is more about what licenses govern the copyrights of the code.  Regardless of by whom and when it was written, the rights to this code is owned by "The Go Authors" in accordance with the [license](https://github.com/protocolbuffers/protobuf-go/blob/master/LICENSE) in its repo, and therefore, "`Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer`".  We don't currently have a way to declare other copyright notices besides our own, or the resources to create a legally compliant way of doing so.

The code discussed here is pretty trivial, but we can surely reimplement similar functionality independently, and do so without directly copying the code.  For instance, I would not have passed an `n` parameter, since it only is used with a single value.  It's also unlikely I'd have chosen the same name for the function, and perhaps wouldn't have made it a separate function at all.

Reply:
The contents of that function really is pretty trivial. 

I think having a function makes sense, so the code isn't duplicated, but I inlined it to show the full scope of what's being done.

The output matches how protoc-gen-go does it.